src/Sensor.cc

```diff
@@ -256,6 +322,9 @@ bool Sensor::Update(const ignition::common::Time &_now,
   // Make the update happen
   result = this->Update(_now);
 
+  // Publish metrics
+  this->PublishMetrics(_now);
```

Comment:
Can we make this optional? We shouldn't add overhead to users who don't care about metrics. Ideally we'd configure this through an SDF element such as `<enable_metrics>`.

If that's too much trouble, for now, I think adding a function like `Sensors::SetEnableMetrics` could be enough, and in the future we configure it from `ign-gazebo`.

Reply:
> We shouldn't add overhead to users who don't care about metrics. Ideally we'd configure this through an SDF element such as <enable_metrics>.

+1. I will follow the approach of enabling it from sdf description.

Reply:
Just to verify. 
Would the correct approach be 
**A:** to modify [`sdf::Sensor`](https://github.com/ignitionrobotics/sdformat/blob/sdf9/include/sdf/Sensor.hh) class to correctly parse the `enable_metrics` field. (even though it affects API) ?
or
**B:** to get the field from sensors::Sensor class by getting the `sdf::ElementPtr` of the `Sensor` and looking for the `enable_metrics` field. ?

I guess `A` is cleaner and is what we are looking for, right @chapulina ?

Reply:
https://github.com/ignitionrobotics/sdformat/pull/665

Reply:
Done!